ceph/ceph_admin/__init__.py

```diff
@@ -88,12 +88,13 @@ def distribute_cephadm_gen_pub_key(self, nodes=None):
 
     def set_tool_repo(self):
         """Add the given repo on every node part of the cluster."""
-        base_url = self.config.get("base_url")
-        if not base_url.endswith("/"):
-            base_url += "/"
-        cmd = f"yum-config-manager --add-repo {base_url}compose/Tools/x86_64/os/"
-        for node in self.cluster.get_nodes():
-            node.exec_command(sudo=True, cmd=cmd)
+        if not self.config.get("hotfix_repo"):
```

Comment:
Is there a possibility of cephadm package not being part of the hotfix-repo? If yes, I believe we should be adding the base url and hot fix repo.

Reply:
After discussing with Rakesh, I understand that `hotfix_repo` would ideally have all the required packages.

Reply:
I think we should always enable `Tools` along with the hotfix repo for the following reasons:
- Customers will already have `rhceph-5-tools-for-rhel-8-x86_64-rpms` enabled on their node(s), and we might as well test in a way that matches what a customer will have.
- It's true that `cephadm` has no non-RHEL dependencies today, so it's probably ok to skip adding `Tools` in that particular case, but we're also planning to ship `ceph-ansible` in `Tools`, and `ceph-ansible` does require some dependencies that are not in RHEL (the deps are only in `Tools`).

Reply:
The issue I ran into was that when I didn't do this the version of `cephadm` from the `Tools` repo was used, not the one from `hotfix_repo`. Perhaps I did something wrong though.

Reply:
On Monday I rebased the downstream packages from `cephadm-16.1.0-100.el8cp` to `cephadm-16.1.0-486.el8cp`. 
Maybe `16.1.0-486.el8cp` in `Tools` was higher than the hotfix repo's cephadm Version-Release?

Reply:
That's exactly what happened, the build I'm testing with is a `16.1.0-100` build. Isn't the point of ``--hotfix-repo`` to be able to test with ceph versions that are potentially lower than the latest available build or release though? I think for our use case we'd always want ``--hotfix-repo`` to take precedence. This is how it works when testing rpms and using ``--hotfix-repo``, iirc.